xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/EditRightsPane.java

```diff
@@ -127,8 +131,12 @@ public void clickGuestRight(Right right)
             // Note: Selenium 2.0a4 returns a relative URL when calling getAttribute("src") but since we moved to
             // Selenium 2.0a7 it returns a *full* URL even though the DOM has a relative URL as in:
             // <img src="/xwiki/resources/js/xwiki/usersandgroups/img/allow.png">
-            getDriver().waitUntilElementEndsWithAttributeValue(buttonLocator, "src",
-                currentState.getNextState().imageURL);
+            getDriver().waitUntilCondition(webDriver -> {
+                // starting with 11.1RC1 the resource URLs can now contain a query parameter to avoid cache issue
+                // so we cannot rely to a full URL anymore
+                return webDriver.findElement(buttonLocator).getAttribute("src")
+                    .contains(currentState.getNextState().imageURL);
+            });
```

Comment:
Why not introduce a waitUntilElementContainsAttributeValue?

Reply:
I hesitated, we have already lot of APIs I think: do you think it might be useful elsewhere too?